### DISCUSSION TITLE
Just simple Q&A about broadcast

### DISCUSSION THREAD

**Discussion by NanaChiki** [Cyfrin/foundry-full-course-cu#940] [Q&A]
I am just wondering  when we actually have to use broadcast commands.
How can we distinguish between when we use 'vm.startbroadcast' and 'vm.stopbroadcast' and when we don't use it.   

Thank you!

**Comment by 0xusmanf** [comment]
By default, Foundry simulates scripts locally, so to tell Foundry to deploy a contract or perform a transaction on-chain(on the blockchain) you have to explicitly tell Foundry to broadcast it on-chain. So `vm.startbroadcast` tells it to start broadcasting the next line or a block of code on-chain until you find `vm.stopbroadcast`. For example in the below script, Foundry will simulate creating `new HelperConfig();` contract, calling `helperConfig.activeNetworkConfig();` functions and returning `(fundMe, helperConfig);` locally and will only broadcast the `new FundMe(priceFeed);` to the blockchain.
```solidity
// SPDX-License-Identifier: UNLICENSED
pragma solidity ^0.8.19;

import {Script} from "forge-std/Script.sol";
import {HelperConfig} from "./HelperConfig.s.sol";
import {FundMe} from "../src/FundMe.sol";

contract DeployFundMe is Script {
    function run() external returns (FundMe, HelperConfig) {
        HelperConfig helperConfig = new HelperConfig(); // This comes with our mocks!
        address priceFeed = helperConfig.activeNetworkConfig();

        vm.startBroadcast();
        FundMe fundMe = new FundMe(priceFeed);
        vm.stopBroadcast();
        return (fundMe, helperConfig);
    }
}
```

**Reply by NanaChiki** [reply]
Thank you so much!!
You mean, I could deploy a part of code on-chain that is like sepolia or mainnet or other specified network?

**Reply by 0xusmanf** [reply]
> Thank you so much!!

You are welcome, always happy to help.
> You mean, I could deploy a part of code on-chain that is like sepolia or mainnet or other specified network?

Yes exactly.

**Reply by NanaChiki** [reply]
That makes cense👍

**Reply by taramakage** [reply]
If I understand correctly, there's no need to add broadcast cheatcode if we just want to simulate it locally?

**Reply by NanaChiki** [reply]
@taramakage 
In the case of the above script, the instance of "**HelperConfig**"  is merely created to get the priceFeed, and is not stored to the blockchain (local, test, main etc...). On the other hand, the instance of  "**FundMe**" is broadcasted and stored to the locally created chain, which is Anvil, because it is used the cheatcode. So if you want to send a transaction on-chain, regardless of it is local chain, you are required to use the cheatcode.

**Reply by taramakage** [reply]
@NanaChiki  

Thanks for your explanation! So we really just want `FundMe` to be deployed on chain regardless of a real testnet or localnet, and `HelperConfig` is more like a piece of code for interaction which helps us get that `priceFeed` contract address. That makes sense.

**Reply by NanaChiki** [reply]
@taramakage 
Yeah totally. No problem!
